internal/gcloud/gcloud.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"time"
+
+	"golang.org/x/oauth2"
+	exec "golang.org/x/sys/execabs"
+)
+
+// config represents the credentials returned by `gcloud config config-helper`.
+type config struct {
+	Credential struct {
+		AccessToken string    `json:"access_token"`
+		TokenExpiry time.Time `json:"token_expiry"`
+	}
+}
+
+func (c *config) Token() *oauth2.Token {
+	return &oauth2.Token{
+		AccessToken: c.Credential.AccessToken,
+		Expiry:      c.Credential.TokenExpiry,
+	}
+}
+
+// Path returns the absolute path to the gcloud command. If the command is not
+// found it returns an error.
+func Path() (string, error) {
+	g := "gcloud"
+	if runtime.GOOS == "windows" {
+		g = g + ".cmd"
+	}
+	return exec.LookPath(g)
+}
+
+// configHelper implements oauth2.TokenSource via the `gcloud config config-helper` command.
+type configHelper struct{}
+
+// Token helps gcloudTokenSource implement oauth2.TokenSource.
+func (configHelper) Token() (*oauth2.Token, error) {
+	gcloudCmd, err := Path()
+	if err != nil {
+		return nil, err
```

Comment:
Do these errors need more context? Will the make sense when boiled up and returned to the user?

Reply:
Here's the error that occurs when a user doesn't have gcloud:

```
Error: exec: "gcloud": executable file not found in $PATH
```

I think that's pretty good.